### ISSUE TITLE
Getting  'TypeError: Cannot read property 'prototype' of undefined' when transforming plan object

### ISSUE THREAD

**Issue by bramkoot** [typestack/class-transformer#114]
When returning a plain object (non-class) while having _some_ custom transformers in my models, I have issues with class transformer 0.1.8 in combination with routing-controllers 0.7.6. 

The error is 'TypeError: Cannot read property 'prototype' of undefined' and the problem seems to be in checking the prototype property of the target which is undefined in the case of a plain object.

https://github.com/typestack/class-transformer/blob/7f2ad5bc196cd368e4264185dbdc289563a872a7/src/metadata/MetadataStorage.ts#L161

A quickfix that works for me; replace `metadatas.filter(meta => target.prototype` with `metadatas.filter(meta => target && target.prototype`

Although this is an easy fix, I'm not sure if this is the root cause of the issue. I tried to dig a bit deeper in the combination of routing-controllers/class-transformer (and Koa) but that took quite a bit of time in the moment that I wanted to finish a feature. 

Switching back to class-transformer 0.1.7 seems to fix the issue.

**Comment by sparebytes** [comment]
That patch fixed the problem for me as well although I don't think that function should ever be called because the property that was being deserialized did not have a `@Type` decorator.

**Comment by mosesoak** [comment]
+1 This crash is happening for me every time I use `@Transform(customFunction)`.

I don't feel like patching this library so I'm going to skip transforming the incoming query parameter and apply workarounds elsewhere. Look forward to a fix on this!

More information: 
* passing query param `user_ids="1"` vs. `user_ids="1"&user_ids="2"` gives a string or an array, which makes handling the param difficult later
* I'm trying to transform the incoming parameter to an array in the single-string case
* I used `@Type(() => String)` and `@Transform(stringOrArray, { toPlainOnly: true })` where `stringOrArray` is my custom transformer that converts the string to an array.
* Besides fixing the bug on this ticket it would be nice to get this type of transform out of the box since it's a common use case to pass multiple individual params and then want to handle them as a list

**Comment by mosesoak** [comment]
here's the stack trace, somewhat simplified. Using 
    "class-transformer": "^0.1.6",
    "class-validator": "^0.7.0",
    "routing-controllers": "0.7.6",

```
TypeError: Cannot read property 'prototype' of undefined
  at MetadataStorage.ts:161:70
  at Array.filter (<anonymous>)
  at MetadataStorage.findMetadatas (src/MetadataStorage.ts:161:48)
  at MetadataStorage.findTransformMetadatas (src/MetadataStorage.ts:46:21)
  at TransformOperationExecutor.applyCustomTransformations (src/TransformOperationExecutor.ts:190:48)
  at _loop_1 (src/TransformOperationExecutor.ts:165:39)
  at TransformOperationExecutor.transform (class-transformer/TransformOperationExecutor.js:172:17)
  at src/TransformOperationExecutor.ts:40:40
  at Array.forEach (<anonymous>)
  at TransformOperationExecutor.transform (src/TransformOperationExecutor.ts:37:30)
  at _loop_1 (src/TransformOperationExecutor.ts:164:43)
  at TransformOperationExecutor.transform (class-transformer/TransformOperationExecutor.js:172:17)
  at ClassTransformer.classToPlain (src/ClassTransformer.ts:19:25)
  at Object.classToPlain (src/index.ts:17:29)
  at ExpressDriver.BaseDriver.transformResult (src/driver/BaseDriver.ts:143:22)
  at ExpressDriver.handleSuccess (src/driver/express/ExpressDriver.ts:248:23)
  at src/RoutingControllers.ts:161:45
  at <anonymous>
  at process._tickDomainCallback (internal/process/next_tick.js:228:7)
```

**Comment by mosesoak** [comment]
Probable duplicate of #28 (opened October 2016... yikes)

**Comment by NoNameProvided** [comment]
I think this has been fixed in #100.

I will leave this issue open for now. After the next release please report it the issue still exists.

**Comment by NoNameProvided** [comment]
Just published `0.1.9`, please test it and report if you still encounter this issue. Thanks!

**Comment by NoNameProvided** [comment]
Closing this as solved.
